File: web/src/app/utils/hata-yardimcisi.ts
// 401 (oturum gecersiz/suresi dolmus) hatasini component-bazli error callback'lerinde tespit
// etmek icin kullanilir - bu durumda auth-interceptor zaten net bir "Oturum süreniz doldu"
// mesaji gosterip kullaniciyi /login'e yonlendiriyor, component'in KENDI genel "Backend API
// çalışıyor mu?" mesajini AYRICA gostermesine gerek yok.
export function oturumHatasiMi(hata: unknown): boolean {
  return typeof hata === 'object' && hata !== null && (hata as { status?: number }).status === 401;
}

File: desktop/SeyirMobil.Desktop/Services/HataYardimcisi.cs
using System.Net;

namespace SeyirMobil.Desktop.Services;

public static class HataYardimcisi
{
    // 401 (oturum gecersiz/suresi dolmus) hatasi icin form-bazli catch bloklarinin KENDI
    // genel "Backend API calisiyor mu?" mesajini gostermesini engellemek icin kullanilir -
    // bu durumda zaten OturumHandler + AracHareketleriForm'daki merkezi handler net bir
    // "Oturum Sona Erdi" mesaji gosterip kullaniciyi login ekranina donduruyor, ikinci/celiskili
    // bir hata kutusuna gerek yok.
    public static bool OturumSuresiDolduMu(Exception ex) =>
        ex is HttpRequestException http && http.StatusCode == HttpStatusCode.Unauthorized;
}
